test/test_scanner.py

```diff
@@ -181,9 +181,7 @@ def test_does_not_scan_symlinks(self):
             finally:
                 os.unlink("non-existant-link")
 
-    def test_cannot_open_file(self):
+    def test_cannot_open_file(self, caplog):
         """ Test behaviour when file cannot be opened """
-        with pytest.raises(InvalidFileError):
-            self.scanner.scan_file(
-                os.path.join(self.mapping_test_dir, "non-existant-file")
-            )
+        self.scanner.scan_file(os.path.join(self.mapping_test_dir, "non-existant-file"))
```

Comment:
This appears to only check one of the paths for triggering InvalidFileError

Reply:
Yeah, I noticed that when I was updating too.  I haven't decided the best way to trigger the other one to add the appropriate test, but probably some sort of permissions error would do the trick.

Reply:
Hm, nope, permissions doesn't work because file returns "no read permission".

Reply:
Okay, I did a few more experiments but I can't seem to trigger that code.  I'm going to file a bug for this test rather than include it in this PR, since I think people would like this fix sooner rather than later.